clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -618,15 +620,21 @@ private ClientRequest request(long now, String node, MetadataRequest metadata) {
         }
 
         /**
-         * Add a metadata request to the list of sends if we can make one
+         * Return true if there's at least one connection establishment is currently underway
          */
-        private void maybeUpdate(long now, Node node) {
-            if (node == null) {
-                log.debug("Give up sending metadata request since no node is available");
-                // mark the timestamp for no node available to connect
-                this.lastNoNodeAvailableMs = now;
-                return;
+        private boolean isAnyNodeConnecting() {
```

Comment:
@hachikuji I wasn't sure if this should be exposed as NetworkClient's method so just narrowed the scope.
Let me know if you had a different thought(like expecting this will be utilized in different place).